drivers/timer/nrf_rtc_timer.c

```diff
@@ -177,9 +182,21 @@ static void rtc_announce_set_next(void)
  */
 void _timer_idle_enter(s32_t sys_ticks)
 {
-	/* Restrict ticks to max supported by RTC without risking overflow. */
+#ifdef CONFIG_TICKLESS_KERNEL
+	if (sys_ticks != K_FOREVER) {
+		/* Need to reprograme timer if current program is smaller*/
+		if (sys_ticks > expected_sys_ticks) {
+			_set_time(sys_ticks);
+		}
+	} else {
+		expected_sys_ticks = 0;
+		/* Set time to largest possile RTC Tick*/
+		_set_time(_get_max_clock_time());
```

Comment:
Yes Vinayak, 

These calls should not impact ISR, 
I am maintaining uniformity as used by Ramesh for Corex/x86 processor.